### PR TITLE
Improve job task remark when success

### PR DESCRIPTION
**Pull request by BrainBalaka**
## Summary
Add additional explanation in the job task remark to make the context more clear

### CODE REVIEW

**Comment by frandi** [comment]
I would like to keep the return value as it is now to allow other components to utilize it as input for other processes. For the remarks, I'd prefer to add a new property in `TaskRunnerResult` to save the customized message. What do you think @BrainBalaka ?

**Comment by BrainBalaka** [comment]
@frandi Agreed. `ReturnValue` does more make sense to be a usable value, instead of a display message. I'll adjust the code.

File: src/Engine/Polyrific.Catapult.Engine.Core/JobTasks/DeployDbTask.cs [hunk 1]
```diff
@@ -53,14 +53,21 @@ public override async Task<TaskRunnerResult> RunMainTask(Dictionary<string, stri
                 return new TaskRunnerResult(result["errorMessage"].ToString(), !TaskConfig.ContinueWhenError);
 
             var databaseLocation = "";
+            var taskRemarks = "";
             if (result.ContainsKey("databaseLocation"))
+            {
                 databaseLocation = result["databaseLocation"].ToString();
+                taskRemarks = $"The database code in {databaseLocation} has been deployed";
```
[the remark this answers is not in the document]
Comment (BrainBalaka) [reply src/Engine/Polyrific.Catapult.Engine.Core/JobTasks/DeployDbTask.cs]: currently the plugin `EntityFrameworkCore` is returning the data dll name. I guess I'll need it to modify it so it returns the database server name instead. Will work on it.

File: src/Engine/Polyrific.Catapult.Engine.Core/JobTasks/BuildTask.cs [hunk 2]
```diff
@@ -54,14 +54,21 @@ public override async Task<TaskRunnerResult> RunMainTask(Dictionary<string, stri
                 return new TaskRunnerResult(result["errorMessage"].ToString(), !TaskConfig.ContinueWhenError);
 
             var outputArtifact = "";
+            var taskRemarks = "";
             if (result.ContainsKey("outputArtifact"))
+            {
                 outputArtifact = result["outputArtifact"].ToString();
+                taskRemarks = $"The code is published in: {outputArtifact}";
```
Comment (frandi) [inline src/Engine/Polyrific.Catapult.Engine.Core/JobTasks/BuildTask.cs]: Let's make it clearer that this is the build artifact: `The build artifact is located at {outputArtifact}.`

File: src/Engine/Polyrific.Catapult.Engine.Core/JobTasks/CloneTask.cs [hunk 3]
```diff
@@ -53,14 +53,21 @@ public override async Task<TaskRunnerResult> RunMainTask(Dictionary<string, stri
                 return new TaskRunnerResult(result["errorMessage"].ToString(), !TaskConfig.ContinueWhenError);
 
             var cloneLocation = "";
+            var taskRemarks = "";
             if (result.ContainsKey("cloneLocation"))
+            {
                 cloneLocation = result["cloneLocation"].ToString();
+                taskRemarks = $"The repository has been cloned in: {cloneLocation}";
```
Comment (frandi) [inline src/Engine/Polyrific.Catapult.Engine.Core/JobTasks/CloneTask.cs]: `cloned to` would be better.

**Comment by frandi** [comment]
merge-pending-remarks


@BrainBalaka when we have `Pending` status in `Merge` task, can we have the remarks to be something like `A pull request has been submitted to {PRLocation}. Please review it, and restart this queue when it's ready to be merged.`

**Comment by BrainBalaka** [comment]
@frandi Related to the remarks when the `Merge` task is `Success`. Can we instead change it into `Pull request {PRNumber} has been merged`? Since the base branch can be modified to be other than `master`, and the data is available on the previous `Push` task.

**Comment by frandi** [comment]
> @frandi Related to the remarks when the `Merge` task is `Success`. Can we instead change it into `Pull request {PRNumber} has been merged`? Since the base branch can be modified to be other than `master`, and the data is available on the previous `Push` task.

Ok 👍

**Comment by BrainBalaka** [comment]
@frandi I've pushed the changes to modify the remarks, including for the `merge` task

**Pull request merged**